modules/imgproc/src/demosaicing.cpp

```diff
@@ -1584,7 +1584,7 @@ class Bayer2RGB_EdgeAware_T_Invoker :
                 {
                     D[0] = S[0];
                     D[1] = (std::abs(S[-1] - S[1]) > std::abs(S[sstep] - S[-sstep]) ? (S[sstep] + S[-sstep] + 1) : (S[-1] + S[1] + 1)) >> 1;
-                    D[2] = (S[-sstep-1] + S[-sstep+1] + S[sstep-1] + S[sstep+1]) >> 2;
+                    D[2] = (S[-sstep-1] + S[-sstep+1] + S[sstep-1] + S[sstep+1] + 2) >> 2;
```

Comment:
To make all code consistent please fix these lines too:
- 1569
- 1571

Also need to check `SIMDInterpolator` implementation.